### ISSUE TITLE
Create some example modules for admin-side Start menu area

### ISSUE THREAD

**Issue by petecooper** [textpattern/textpattern#1735]
### Is your feature request related to a problem?

Nope.

### What is the feature?

An admin-side dashboard. A landing page with an overview of the admin-side.

Considerations:

* snap-in modules with various insights & traffic light status indicators, e.g.:
  * number of articles, with breakdown for each status – especially helpful for multi-user sites with article approval workflows
  * recent referrers from logs
  * number of comments, with # in moderation queue
  * number of files / images, with any issues highlighted
  * number of items in respective categories
  * brief diagnostics
 * open up to plugin authors
 * introduce as a non-default landing page for admin-side, make default after a release series when bugs are squashed

There's probably a heap more stuff available, really a case of seeing if this has legs and going from there.

**Comment by Bloke** [comment]
This is what the (hidden) Start menu area is for. Normally we leave it up to plugins to fill in such functionality.

Nothing to stop us building such a dashboard and bundling it with core but I wonder if it might be better if we introduced a handy way for plugins to register modules to include on some core-sanctioned panel. That way, we leave implementation to plugins and if admins want a particular stat on their dashboard, just add the plugin(s) that provide these items. Then to turn them on/off is as simple as enable/disable. Uninstall the plugin to remove.

Adding one or more plugins that registers with a dashboard callback automatically populates the Dashboard panel. Even today, that panel will get added (again, automatically) to the list of available start panels in prefs so if someone wants to make it their default panel, that's covered without any additional code.

In core, all we need is:

* A callback to allow plugins to register a module (in plugin ext_some_dash_feature.php).
* A standard dashboard page that has a flex grid on it, with drag/drop facility.
* A change to the menu code so if there are any plugins registered in the dashboard 'pre' step (arbitrary 'pre' is already baked in) the dashboard page is included.
* If anyone drags and drops a module to another location in the grid, remember the grid order in localStorage or hidden per-user pref.

Job done.

In terms of output, a plugin could output whatever it likes BUT to make it easy, if we introduce a convention that plugins can return an array, or class of type 'Module', with key field data completed (at minimum, module title, and module HTML or list or something) then it'll be rendered in a standard way in its grid cell. If any plugin author wants to bypass that, fine, just output content as normal using `return 'some html here';` and it'll be dropped in their grid cell.

A single plugin can register multiple modules by raising the same callback with a different function endpoint. So a plugin author can offer a suite of modules and even elect to allow people to turn them on and off via prefs. Alternatively, the dashboard code in core could expose prefs on authors' behalf. Introduce a 'Dashboard' area in prefs and populate it with a series of yes/no radios - one per registered dash callback - so all dashboard elements can be made individually available by an admin to all users.

Another option is to allow each grid cell to be deleted via an 'x' in its corner, whereby it drops into an 'available' drawer somewhere on-screen, which you can pop out and drag the modules into place on the grid.

tl;dr, plenty of ways we could approach this. But I'd rather build a modular system to allow plugins to add pockets of functionality than offer actual dash elements in core. Such plugins can include core-sanctioned modules in, e.g., com_dashboard, which would allow us to supply a suite of handy modules without them actually being part of core itself.

Anyone got any more ideas around this?

**Comment by petecooper** [comment]
Ack. I should have done my homework here.

I agree with everything above - optional modules is absolutely fine, and a few examples to get the ball rolling (at whatever pace that happens) would be perfect.

**Comment by Bloke** [comment]
Cool, so let's leave this open to track it then. I can look at this and see what we can do here.

**Comment by petecooper** [comment]
>This is what the (hidden) Start menu area is for. Normally we leave it up to plugins to fill in such functionality.

OP sorted, closing. Can be deferred to plugins for now.
